test/rust_src/src/keymaps-test.el

```diff
@@ -0,0 +1,21 @@
+(require 'ert)
+
+(ert-deftest keymap-local-tests ()
+  (let ((sample-keymap '(keymap
+                         (3 keymap
+                            ;; C-c C-z
+                            (26 . emacs-version)))))
+    (use-local-map nil)
+    (should-not (current-local-map))
+    (use-local-map sample-keymap)
+    (should (equal (current-local-map) '(keymap (3 keymap (26 . emacs-version)))))))
+
+(ert-deftest keymap-global-tests ()
+  (let ((sample-keymap '(keymap
+                         (3 keymap
+                            ;; C-c C-z
+                            (26 . emacs-version))))
+        (backup-keymap (current-global-map)))
+    (use-global-map sample-keymap)
+    (should (equal (current-global-map) '(keymap (3 keymap (26 . emacs-version)))))
+    (use-global-map backup-keymap)))
```

Comment:
You should also test what happens when NIL is passed like you do in `use-local-keymap` tests.

Reply:
I tried it out and it seems the original c function errors when passed with `nil`. So it seems I might need to handle `nil` on my ported function.

In situations when you find bugs in Emacs and fixed it in Remacs, do you guys also submit a fix to Emacs?

Reply:
`(wrong-type-argument keymapp nil)` is exactly the error the C code says should happen when a nil is passed. The C code does not have a bug.

Look at roughly line 37 of the function `get_keymap` in keymap.c.

Reply:
That said, when we find bugs we send them upstream. If you want to include a patch that is up to you. Including a test in your patch is a good idea.

Reply:
Aha! I should have checked where the error was coming from. This means the check `(should-error (use-global-map nil))` should do the trick. Will be fixing and updating my pull request in a bit. Thanks!